java/client/src/main/java/glide/api/models/commands/WeightAggregateOptions.java

```diff
@@ -0,0 +1,64 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import glide.api.commands.SortedSetBaseCommands;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.Singular;
+
+/**
+ * Optional arguments to {@link SortedSetBaseCommands#zunionstore}
+ *
+ * @see <a href="https://redis.io/commands/zunionstore/">redis.io</a> for more details.
+ */
+@Builder
+public final class WeightAggregateOptions {
+    /**
+     * Represents multiplication factors for each sorted set, ready for aggregation. Each
+     * multiplication factor corresponds one-to-one with the sets. The score of every element in these
+     * sets is multiplied by its associated factor before aggregation.
+     */
+    @Singular private final List<Double> weights;
+
+    /**
+     * Option for the method of aggregating scores from multiple sets. This option defaults to SUM if
+     * not specified.
+     */
+    private final Aggregate aggregate;
+
+    /**
+     * Option for the method of aggregating scores from multiple sets. This option defaults to SUM if
+     * not specified.
+     */
+    public enum Aggregate {
+        /** Aggregates by summing the scores of each element across sets. */
+        SUM,
+        /** Aggregates by selecting the minimum score for each element across sets. */
+        MIN,
+        /** Aggregates by selecting the maximum score for each element across sets. */
+        MAX;
+    }
+
+    /**
+     * Converts WeightAggregateOptions into a String[].
+     *
+     * @return String[]
+     */
+    public String[] toArgs() {
+        List<String> optionArgs = new ArrayList<>();
+
+        if (!weights.isEmpty()) {
```

Comment:
Would this blow up if weights is null?

Reply:
Nope its already initialized with singular.